util/signing/interface/src/cryptography/mod.rs

```diff
@@ -29,15 +33,18 @@ macro_rules! fixed_size {
 pub mod ed25519;
 pub mod secp256k1;
 
-pub trait TryFromBytes: Sized {
-	fn try_from_bytes(bytes: &[u8]) -> Result<Self, anyhow::Error>;
-}
+use std::error::Error;
 
 /// A designator for an elliptic curve.
 ///
 /// This trait has no methods, but it binds the types of the public key and
 /// the signature used by the EC digital signing algorithm.
 pub trait Curve {
-	type PublicKey: TryFromBytes;
+	type PublicKey;
 	type Signature;
 }
+
+/// Errors that occur when parsing signature or key material from byte sequences.
+#[derive(Debug, thiserror::Error)]
+#[error(transparent)]
+pub struct CryptoMaterialError(Box<dyn Error + Send + Sync>);
```

Comment:
I think it will be easier to have one error type for the crate. It avoids adding map_err every time when you need to extract the publickey from an u8 array, for example, and it eases the error management for external crate.
There's a reason you didn't add a new variant to the SignerError. Perhaps we need to rename it in a more generic name?